core/asm/asm.go

```diff
@@ -66,7 +66,7 @@ func (it *instructionIterator) Next() bool {
 
 	it.op = vm.OpCode(it.code[it.pc])
 	if it.op.IsPush() {
-		a := uint64(it.op) - uint64(vm.PUSH1) + 1
+		a := uint64(it.op) + 1 - uint64(vm.PUSH1)
```

Comment:
I don't think it's wrong here, although I get your point.

I will vote another solution

```diff
diff --git a/core/asm/asm.go b/core/asm/asm.go
index 294eb6ffaa..ff41ff5315 100644
--- a/core/asm/asm.go
+++ b/core/asm/asm.go
@@ -66,7 +66,7 @@ func (it *instructionIterator) Next() bool {
 
        it.op = vm.OpCode(it.code[it.pc])
        if it.op.IsPush() {
-               a := uint64(it.op) - uint64(vm.PUSH1) + 1
+               a := uint64(it.op) - uint64(vm.PUSH0)
                u := it.pc + 1 + a
                if uint64(len(it.code)) <= it.pc || uint64(len(it.code)) < u {
                        it.error = fmt.Errorf("incomplete push instruction at %v", it.pc)

```

It's easier to understand how the amount of arguments is calculated.

Reply:
yes